tests/js/common/shell/arangosearch-update-consistency.js

```diff
@@ -54,9 +76,169 @@ function UpdateConsistency() {
+      }
+    }
+  };
+}
+
+function UpdateConsistencySearchAliasView() {
+  const collection = "testUpdate";
+  const view = "testUpdateView";
+  return {
+    setUp: function () {
+      let collectionProperties = {};
+      if (isCluster) {
+        collectionProperties = {numberOfShards: 9, replicationFactor: 3};
+      }
+      let c = db._create(collection, collectionProperties);
+      c.save({_key: "279974", value: 345, flag: true});
+      c.ensureIndex({name: "inverted_index", type: "inverted", fields: ["value"]});
+      v = db._createView(view, "search-alias", {});
+    },
+
+    tearDown: function () {
+      db._dropView(view);
+      db._drop(collection);
+    },
+
+    testInvertedSingleUpdateLinkedField: function () {
+      let c = db._collection(collection);
+      v = db._view(view);
+      v.properties({indexes: [{collection: "testUpdate", index: "inverted_index"}]});
+
+      for (let j = 0; j < 100; j++) {
+        let oldLen = db._query("FOR d IN " + view + " SEARCH d.value == 345 RETURN d").toArray().length;
+        assertEqual(oldLen, 1);
+        c.update("279974", {value: 345, flag: true});
+        let newLen = db._query("FOR d IN " + view + " SEARCH d.value == 345 RETURN d").toArray().length;
+        assertEqual(newLen, 1);
+        c.update("279974", {value: 345, flag: false});
+      }
+    },
+
+    testInvertedSingleleUpdateUnlinkedField: function () {
+      let c = db._collection(collection);
+      v = db._view(view);
+      v.properties({indexes: [{collection: "testUpdate", index: "inverted_index"}]});
+
+      for (let j = 0; j < 100; j++) {
+        let oldLen = db._query("FOR d IN " + view + " SEARCH d.value == 345 RETURN d").toArray().length;
+        assertEqual(oldLen, 1);
+        c.update("279974", {unlinkedField: j, flag: true});
+        let newLen = db._query("FOR d IN " + view + " SEARCH d.value == 345 RETURN d").toArray().length;
+        assertEqual(newLen, 1);
+        c.update("279974", {unlinkedField: j, flag: false});
+      }
+    },
+
+    testMultiUpdates: function () {
+      let collectionProperties = {};
+      if (isCluster) {
+        collectionProperties = {numberOfShards: 9, replicationFactor: 3};
+      }
+
+      const amountOfCollections = 3;
+      const amountOfDocuments = 10;
+      const documents = [];
+
+      for (let i = 0; i < amountOfDocuments; i++) {
+        documents.push({_key: `doc${i}`, value: i});
+      }
+
+      v = db._view(view);
+      const insertedDocs = [];
+      for (let i = 0; i < amountOfCollections; i++) {
+        c = db._create(`testUpdate${i}`, collectionProperties);
+        c.ensureIndex({name: "inverted_index", type: "inverted", fields: ["value"]});
+        insertedDocs.push(db[`testUpdate${i}`].insert(documents));
+        v.properties({indexes: [{collection: `testUpdate${i}`, index: "inverted_index"}]});
+      }
+
+      // wait for search-alias view index initialization
+      db._query("RETURN SLEEP(2)")
```

Comment:
we have waitForSync

Reply:
Removed `SLEEP()` and added `OPTIONS { waitForSync: true }` query